src/ex2/Uilogic.js

```diff
@@ -0,0 +1,444 @@
+const TYPE_DEADLINE = 1
+const TYPE_NOTCOMPLETE = 2
+const TYPE_FINISHED = 3
+const TYPE_DEADLINE_FINISHED = 4
+const SORT_ALPHABATICALLY = 1;
+const SORT_DEADLINE = 2;
+const SORT_NOTCOMPLETE = 3;
+const SORT_FINISHED = 4;
+
+class UiLogic {
+  constructor() {
+    this.taskList = [];
+  }
+
+  renderItem(taskList) {
+    this.addTaskWatcher();
+    const filterTaskList = taskList.filter((element) => element.isDisplay === false)
+    filterTaskList.forEach(element => {
+      let taskName;
+      if (element.isPokemon === false) {
+        taskName = element.item
+        element.isDisplay = true;
+      } else {
+        taskName = element.item.name;
+        element.isDisplay = true;
+      }
+      const isPokemon = element.isPokemon
+      this.CreateNewListItemElement(taskName, taskList, isPokemon);
+    });
+  }
+
+  addTaskWatcher() {
+    setInterval(() => {
+      let deadLineReachedTask = this.taskList.filter(x => x.type == TYPE_DEADLINE).filter(x => x.datetime < new Date());
+      if (deadLineReachedTask && deadLineReachedTask.length > 0) {
+        let ids = deadLineReachedTask.map(x => x.id);
+        for (let index = 0; index < ids.length; index++) {
+          const liId = ids[index];
+          let task = getTask(liId);
+          this.updateTask(task.id, TYPE_DEADLINE_FINISHED, task.name, task.datetime, task.isAnimated);
+          this.bindTaskList(false);
+
+        }
+      }
+
+    }, 1000)
+  }
+
+  addTask(type, name, date, isPokemon) {
+    let id = this.getId();
+    if (isPokemon === true) {
+      name = "Catch " + name;
+    }
+    this.taskList.push({ id: id, type: type, name: name, datetime: date, createDateTime: new Date(), isAnimated: true });
+    return id;
+  }
+  CreateNewListItemElement(taskString, taskList, isPokemon) {
+    if (taskString === '') {
+      alert("You must write something!");
+    } else {
+      this.addTask(TYPE_NOTCOMPLETE, taskString, null, isPokemon);
+      this.bindTaskList(false, taskList)
+    }
+  }
+
+  getId() {
+    if (this.taskList.length == 0) {
+      return 1;
+    }
+    const latestId = this.taskList.map(x => x.id).sort((p, c) => c - p);
+    return latestId[0] + 1;
+
+  }
+  getTask(id) {
+    return this.taskList.filter(x => x.id == id)[0]
+  }
+
+  updateTask(id, type, name, date, isAnimated) {
+    let indexFound = 0;
+    let createDateTime = new Date();
+    for (let index = 0; index < this.taskList.length; index++) {
+      let element = this.taskList[index];
+      if (element.id == id) {
+        createDateTime = element.createDateTime;
+        indexFound = index;
+        break;
+      }
+    }
+
+    this.taskList[indexFound] = { id: id, type: type, name: name, datetime: date, createDateTime: createDateTime, isAnimated: isAnimated };
+  }
+
+  bindTaskList(withSort, taskList) {
```

Comment:
consider: it makes more sense to get the taskList as a first argument, because it's mandatory, and have the second one as optional (with default value)
```suggestion
  bindTaskList(taskList, withSort = false) {
```